### PR TITLE
Warn on intitialization of Simple SpanProcessor

### LINKED ISSUE #1875: Warn on intitialization of Simple SpanProcessor
### Problem Statement

The `simpleSpanProcessor` is not intended for use outside of development and debugging.  However, many early examples and demos included its use and users report with some regularity performance issues caused by its use.  

### Proposed Solution

We should emit a warning when `NewSimpleSpanProcessor()` is invoked.

#### Alternatives

Documentation and examples have already been updated to use a batch span processor.

### PR DESCRIPTION
**Pull request by dubonzi**
Ended up not creating a `Warn` function as mentioned in the issue since `Info` is already being used for warning messages in other places.

Resolves #1875

### CODE REVIEW

File: sdk/trace/simple_span_processor.go [hunk 1]
```diff
@@ -43,6 +44,8 @@ func NewSimpleSpanProcessor(exporter SpanExporter) SpanProcessor {
 	ssp := &simpleSpanProcessor{
 		exporter: exporter,
 	}
+	global.Info("SimpleSpanProcessor is not recommended for production use, consider using BatchSpanProcessor instead")
```
Comment (dmathieu) [inline sdk/trace/simple_span_processor.go:47]:
```suggestion
	global.Info("SimpleSpanProcessor is not recommended for production use, consider using BatchSpanProcessor instead.")
```

**Review by dmathieu — COMMENTED** [review]
This should have a changelog entry.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
+- Added a log message warning about usage of `SimpleSpanProcessor` in production environments. (#1875)
```
Comment (dmathieu) [inline CHANGELOG.md:14]: This should be the number of the PR fixing the issue, not the related issue.

```suggestion
- Added a log message warning about usage of `SimpleSpanProcessor` in production environments. (#3854)
```

Comment (dubonzi) [reply CHANGELOG.md]: Ah, my bad.

**Review by dmathieu — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by dubonzi** [comment]
I just rebased and switched to using the new `Warn` function added in #3900.

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**